tests/integration/modules/mac_ports.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+'''
+integration tests for mac_ports
+'''
+
+# Import python libs
+from __future__ import absolute_import, print_function
+
+# Import Salt Testing libs
+from salttesting.helpers import ensure_in_syspath, destructiveTest
+ensure_in_syspath('../../')
+
+# Import salt libs
+import integration
+import salt.utils
+
+
+class MacPortsModuleTest(integration.ModuleCase):
+    '''
+    Validate the mac_ports module
+    '''
+    AGREE_INSTALLED = False
+
+    def setUp(self):
+        '''
+        Get current settings
+        '''
+        if not salt.utils.is_darwin():
+            self.skipTest('Test only available on Mac OS X')
+
+        if not salt.utils.which('port'):
+            self.skipTest('Test requires port binary')
+
+        if salt.utils.get_uid(salt.utils.get_user()) != 0:
+            self.skipTest('Test requires root')
+
+        self.AGREE_INSTALLED = 'agree' in self.run_function('pkg.list_pkgs')
+        self.run_function('pkg.refresh_db')
+
+    def tearDown(self):
+        '''
+        Reset to original settings
+        '''
+        if not self.AGREE_INSTALLED:
+            self.run_function('pkg.remove', ['agree'])
```

Comment:
Also, this won't restore us to original settings. If a person had this on their system and needed it, they'll be left without it.